### ISSUE TITLE
Can`t see the collection on my opensea profile!

### ISSUE THREAD

**Issue by Jam6w** [hashlips-lab/nft-erc721-collection#182]
Hey, Hashlips Hey liarco..

I had deploy the contract and verify it, then open public sale on polygon network by truffle dashboard but I didn`t see the collection on opensea on my profile, I tried to import the contract and I get this error"We couldn't find this contract. Please ensure that this is a valid ERC721 or ERC1155 contract deployed on Polygon and that you have already minted items on the contract."
and I tried " yarn test" and it gave me that:

 ✔ Contract deployment (1005ms)
    ✔ Check initial data (450ms)
    ✔ Before any sale (528ms)
    ✔ Whitelist sale (513ms)
    ✔ Pre-sale (same as public sale) (292ms)
    ✔ Owner only functions (344ms)
    ✔ Wallet of owner (150ms)
    - Supply checks (long)
    ✔ Token URI generation (176ms)

  8 passing (41s)
  1 pending

Any help with this issue..

Thanks..

**Comment by gokugen** [comment]
Hi me too,
I've deployed the contract on the testnet (rinkeby) network and verify it but when I mint an NFT on the website I can't see it on my OpenSea's account. I used the same network and account for all the operation.
The transactions are displayed on etherscan so I don't really understand. Maybe it's a problem with the metadata?

Thank you for your time

**Comment by gokugen** [comment]
Actually I can now see them. It did take some time in order to be updated...

@Jam6w maybe you just need to wait a little bit more.

**Comment by dcryptodev** [comment]
Hi @liarco 

Im facing the same exact issue. My smart contract is working fine on mumbai.polygonscan.com. I went through all the steps and everything worked well. I went to testnet Mumbai on Opensea to upload the smart contract and it gave me the same error as @Jam6w 

`We couldn't find this contract. Please ensure that this is a valid ERC721 or ERC1155 contract deployed on Mumbai and that you have already minted items on the contract.` 

I've already minted items on the contract. @liarco will really appreciate your help here. 

@gokugen the issue myself and @Jam6w are having is actually uploading the smart contract at step two when importing smart contract to opensea https://testnets.opensea.io/get-listed/step-two

**Comment by ShanukJ** [comment]
I'm having the same issue.

**Comment by Salro1g** [comment]
I have the same issue, but from yesterday. Before this, I could see all my collections so I think the problem is on Opensea, not in the code. Open sea is now implemeting Solana NFTs in their place market, so maybe the problem is cause by this. Any suggestions?

**Comment by Salro1g** [comment]
I know that I have all my metadata correct, why do I know? Because I could see few NFTs from the collection but not all minted. And it was unrevealed so the metadata is only 1. For this reason, I have tried delete de collection from Opensea and turn to create but now is telling me:

We couldn't find this contract. Please ensure that this is a valid ERC721 or ERC1155 contract deployed on Rinkeby and that you have already minted items on the contract.

**Comment by ShanukJ** [comment]
@Salro1g I got the same error yesterday. I also deleted the collection and redeployed it. There can be some issues with CID, but I can't guarantee that. Can you see your collection on rarible? If you can see it on rarible, it will be available in opensea soon. Don't mint the whole collection once too.

**Comment by Jam6w** [comment]
@ShanukJ hey man..

How we can mint some of the collection not all of it.

Thank you..

**Comment by ecerroni** [comment]
> @Salro1g I got the same error yesterday. I also deleted the collection and redeployed it. There can be some issues with CID, but I can't guarantee that. Can you see your collection on rarible? If you can see it on rarible, it will be available in opensea soon. Don't mint the whole collection once too.

I have actually minted my collection items all at once and it never showed up in Opensea while I was able to import it on Rarible.

So, given your advice, I am curious if you know the reason of why it's not wise to mint all nft at once :)

**Comment by liarco** [comment]
OpenSea can be really slow updating collections, this happens on the mainnet, so be sure it can take days on the testnets.

Make sure your metadata is correct and rely on what's on the chain: if you can see your tokens on the chain and your metadata is fine, then there is nothing more you can do... it should be an issue on OpenSea side.

> So, given your advice, I am curious if you know the reason of why it's not wise to mint all nft at once :)

@ecerroni there is no rule here, but a single mint operation with thousands of tokens can take a huge time for OpenSea to sync up. I don't know the exact reason why this happens, maybe they worked on some optimizations assuming nobody would mint so many tokens at once. But I might be wrong here, this is just based on my own experience.

**Comment by ShanukJ** [comment]
Hey @Jam6w, as @liarco  mentioned, it will take a massive amount of time to show up the collection in opensea if you minted the whole collection using one single mint operation. I tested it. I minted a 10k collection using a single mint operation, and it went well. I got the success logs and all transaction logs. But it took several hours to show up my collection in opensea. But some of the NFTs were missing. For example, 1 to 1000 was there and 1000 to 2000 missing. I deployed another smart contract to prevent it and minted my collection as 1000 batches. First, mint the 1000 and wait to show them up in opensea. It will take a few minutes. Once you see all of them, mint another 1000. That's how I managed to speed up the process and minted 10k. I hope this helps.
